Dashboard/app/js/lib/views/views-public.js

```diff
@@ -52,13 +52,85 @@ Ember.Handlebars.registerHelper('tooltip', function (i18nKey) {
   );
 });
 
+function renderCaddisflyAnswer(json){
+  name = ""
+  imageUrl = ""
+  result = Ember.A();
+  if (!Ember.empty(json)){
+    jsonParsed = JSON.parse(json);
+
+    // get out test name
+    if (!Ember.empty(jsonParsed.name)){
+      testName =  jsonParsed.name.trim();
+    }
+
+    // get out results
+    if (jsonParsed.result && !Ember.empty(jsonParsed.result)){
+        jsonParsed.result.forEach(function(item){
+        newResult = {
+          "name":item.name,
+          "value":item.value,
+          "unit":item.unit,
+        };
+        result.push(newResult);
+      });
+    }
+    // get out image url
+    if (!Ember.empty(jsonParsed.image)){
+      imageUrl = FLOW.Env.photo_url_root + jsonParsed.image.trim();
+    }
+
+    // contruct html
+    html = "<div><strong>" + name + "</strong></div>"
+    result.forEach(function(item){
```

Comment:
It seems that we're inconsistently use `forEach` or `map` + `join`. It feels more _clean_ to use map + join then modifying a closed variable `html` in the outer scope.